### PR TITLE
Added return value decorator

### PR DESCRIPTION
**Pull request by aweeks**

### CODE REVIEW

File: metricslogging/metricslogging.py [hunk 1]
```diff
@@ -311,6 +312,20 @@ def foo():
         """
         return CounterContextDecorator(self, name, sample_rate)
 
+    def return_val_gauge_d(self, name):
```
Comment (jimrollenhagen) [inline metricslogging/metricslogging.py]: I don't love the name, but it's fine

**Comment by jimrollenhagen** [comment]
Seems fine

**Comment by aweeks** [comment]
@jimrollenhagen  

shipit?

**Pull request merged**